Minimum supported version tests failing

Following merging of PR #1909, the 

https://github.com/scikit-hep/pyhf/blob/f01dcf59da10ec9313345305efc49cfb7186bf99/.github/workflows/lower-bound-requirements.yml#L1

workflow started failing for

https://github.com/scikit-hep/pyhf/blob/f01dcf59da10ec9313345305efc49cfb7186bf99/tests/test_scripts.py#L166

with

```
=================================== FAILURES ===================================
______________________ test_import_usingMounts[inprocess] ______________________

datadir = PosixPath('/tmp/pytest-of-runner/pytest-0/test_import_usingMounts_inproc0')
tmpdir = local('/tmp/pytest-of-runner/pytest-0/test_import_usingMounts_inproc0')
script_runner = <ScriptRunner inprocess>

    def test_import_usingMounts(datadir, tmpdir, script_runner):
        data = datadir.joinpath("xmlimport_absolutePaths")
    
        temp = tmpdir.join("parsed_output.json")
        command = f'pyhf xml2json --hide-progress -v {data}:/absolute/path/to -v {data}:/another/absolute/path/to --output-file {temp.strpath:s} {data.joinpath("config/example.xml")}'
    
        ret = script_runner.run(*shlex.split(command))
>       assert ret.success
E       assert False
E        +  where False = <pytest_console_scripts.RunResult object at 0x7f93d[84](https://github.com/scikit-hep/pyhf/runs/8031258926?check_suite_focus=true#step:6:85)a6750>.success

command    = 'pyhf xml2json --hide-progress -v /tmp/pytest-of-runner/pytest-0/test_import_usingMounts_inproc0/xmlimport_absolutePat..._output.json /tmp/pytest-of-runner/pytest-0/test_import_usingMounts_inproc0/xmlimport_absolutePaths/config/example.xml'
data       = PosixPath('/tmp/pytest-of-runner/pytest-0/test_import_usingMounts_inproc0/xmlimport_absolutePaths')
datadir    = PosixPath('/tmp/pytest-of-runner/pytest-0/test_import_usingMounts_inproc0')
ret        = <pytest_console_scripts.RunResult object at 0x7f93d84a6750>
script_runner = <ScriptRunner inprocess>
temp       = local('/tmp/pytest-of-runner/pytest-0/test_import_usingMounts_inproc0/parsed_output.json')
tmpdir     = local('/tmp/pytest-of-runner/pytest-0/test_import_usingMounts_inproc0')

tests/test_scripts.py:173: AssertionError
----------------------------- Captured stdout call -----------------------------
# Running console script: pyhf xml2json --hide-progress -v /tmp/pytest-of-runner/pytest-0/test_import_usingMounts_inproc0/xmlimport_absolutePaths:/absolute/path/to -v /tmp/pytest-of-runner/pytest-0/test_import_usingMounts_inproc0/xmlimport_absolutePaths:/another/absolute/path/to --output-file /tmp/pytest-of-runner/pytest-0/test_import_usingMounts_inproc0/parsed_output.json /tmp/pytest-of-runner/pytest-0/test_import_usingMounts_inproc0/xmlimport_absolutePaths/config/example.xml
# Script return code: 1
# Script stdout:

# Script stderr:
Traceback (most recent call last):
  File "/opt/hostedtoolcache/Python/3.7.13/x64/lib/python3.7/site-packages/pytest_console_scripts.py", line 188, in console_script
    return s()
  File "/opt/hostedtoolcache/Python/3.7.13/x64/lib/python3.7/site-packages/click/core.py", line 764, in __call__
    return self.main(*args, **kwargs)
  File "/opt/hostedtoolcache/Python/3.7.13/x64/lib/python3.7/site-packages/click/core.py", line 717, in main
    rv = self.invoke(ctx)
  File "/opt/hostedtoolcache/Python/3.7.13/x64/lib/python3.7/site-packages/click/core.py", line 1137, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/opt/hostedtoolcache/Python/3.7.13/x64/lib/python3.7/site-packages/click/core.py", line 956, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/opt/hostedtoolcache/Python/3.7.13/x64/lib/python3.7/site-packages/click/core.py", line 555, in invoke
    return callback(*args, **kwargs)
  File "/opt/hostedtoolcache/Python/3.7.13/x64/lib/python3.7/site-packages/pyhf/cli/rootio.py", line 63, in xml2json
    validation_as_error=validation_as_error,
  File "/opt/hostedtoolcache/Python/3.7.13/x64/lib/python3.7/site-packages/pyhf/readxml.py", line 455, in parse
    ET.parse(resolver(inp)), resolver, track_progress
  File "/opt/hostedtoolcache/Python/3.7.13/x64/lib/python3.7/xml/etree/ElementTree.py", line 1197, in parse
    tree.parse(source, parser)
  File "/opt/hostedtoolcache/Python/3.7.13/x64/lib/python3.7/xml/etree/ElementTree.py", line 5[87](https://github.com/scikit-hep/pyhf/runs/8031258926?check_suite_focus=true#step:6:88), in parse
    source = open(source, "rb")
FileNotFoundError: [Errno 2] No such file or directory: '/absolute/path/to/config/example_channel.xml'
```

we had thought that PR #1917 would fix this, but it hasn't. Given the core requirements of pyhf are modest

https://github.com/scikit-hep/pyhf/blob/f01dcf59da10ec9313345305efc49cfb7186bf99/tests/constraints.txt#L1-L9

it is unclear to me why the tests with these unpinned pass, but fail here.

Running 

```console
$ pytest tests/test_scripts.py -k 'test_import_usingMounts'            
```

fails

```
================================================================================= test session starts =================================================================================
platform linux -- Python 3.10.4, pytest-7.1.2, pluggy-1.0.0
Matplotlib: 3.5.2
Freetype: 2.6.1
benchmark: 3.4.1 (defaults: timer=time.perf_counter disable_gc=False min_rounds=5 min_time=0.000005 max_time=1.0 calibration_precision=10 warmup=False warmup_iterations=100000)
rootdir: /home/feickert/Code/GitHub/pyhf, configfile: pyproject.toml
plugins: console-scripts-1.3.1, mpl-0.16.0, benchmark-3.4.1, cov-3.0.0, mock-3.8.2, requests-mock-1.9.3, anyio-3.6.1
collected 62 items / 60 deselected / 2 selected                                                                                                                                       

tests/test_scripts.py F.                                                                                                                                                        [100%]

====================================================================================== FAILURES =======================================================================================
_________________________________________________________________________ test_import_usingMounts[inprocess] __________________________________________________________________________

datadir = PosixPath('/tmp/pytest-of-feickert/pytest-4/test_import_usingMounts_inproc0'), tmpdir = local('/tmp/pytest-of-feickert/pytest-4/test_import_usingMounts_inproc0')
script_runner = <ScriptRunner inprocess>

    def test_import_usingMounts(datadir, tmpdir, script_runner):
        data = datadir.joinpath("xmlimport_absolutePaths")
    
        temp = tmpdir.join("parsed_output.json")
        command = f'pyhf xml2json --hide-progress -v {data}:/absolute/path/to -v {data}:/another/absolute/path/to --output-file {temp.strpath:s} {data.joinpath("config/example.xml")}'
    
        ret = script_runner.run(*shlex.split(command))
        assert ret.success
        assert ret.stdout == ''
>       assert ret.stderr == ''
E       AssertionError: assert 'DEBUG:pyhf.c...output.json\n' == ''
E         + DEBUG:pyhf.cli.rootio:Written to /tmp/pytest-of-feickert/pytest-4/test_import_usingMounts_inproc0/parsed_output.json

command    = 'pyhf xml2json --hide-progress -v /tmp/pytest-of-feickert/pytest-4/test_import_usingMounts_inproc0/xmlimport_absoluteP...utput.json /tmp/pytest-of-feickert/pytest-4/test_import_usingMounts_inproc0/xmlimport_absolutePaths/config/example.xml'
data       = PosixPath('/tmp/pytest-of-feickert/pytest-4/test_import_usingMounts_inproc0/xmlimport_absolutePaths')
datadir    = PosixPath('/tmp/pytest-of-feickert/pytest-4/test_import_usingMounts_inproc0')
ret        = <pytest_console_scripts.RunResult object at 0x7fe02f776740>
script_runner = <ScriptRunner inprocess>
temp       = local('/tmp/pytest-of-feickert/pytest-4/test_import_usingMounts_inproc0/parsed_output.json')
tmpdir     = local('/tmp/pytest-of-feickert/pytest-4/test_import_usingMounts_inproc0')

tests/test_scripts.py:175: AssertionError
-------------------------------------------------------------------------------- Captured stdout call ---------------------------------------------------------------------------------
# Running console script: pyhf xml2json --hide-progress -v /tmp/pytest-of-feickert/pytest-4/test_import_usingMounts_inproc0/xmlimport_absolutePaths:/absolute/path/to -v /tmp/pytest-of-feickert/pytest-4/test_import_usingMounts_inproc0/xmlimport_absolutePaths:/another/absolute/path/to --output-file /tmp/pytest-of-feickert/pytest-4/test_import_usingMounts_inproc0/parsed_output.json /tmp/pytest-of-feickert/pytest-4/test_import_usingMounts_inproc0/xmlimport_absolutePaths/config/example.xml
# Script return code: 0
# Script stdout:

# Script stderr:
DEBUG:pyhf.cli.rootio:Written to /tmp/pytest-of-feickert/pytest-4/test_import_usingMounts_inproc0/parsed_output.json
```

even though running the full scripts tests suite passes

```console
$ pytest tests/test_scripts.py
```

so it might additionally be worth considering checking if

https://github.com/scikit-hep/pyhf/blob/f01dcf59da10ec9313345305efc49cfb7186bf99/tests/test_scripts.py#L175

is still actually needed/desirable.

--- a/setup.cfg
+++ b/setup.cfg
@@ -36,7 +36,7 @@ include_package_data = True
 python_requires = >=3.7
 install_requires =
     scipy>=1.1.0 # requires numpy, which is required by pyhf and tensorflow
-    click>=7.0  # for console scripts
+    click>=8.0.0  # for console scripts
     tqdm>=4.56.0  # for readxml
     jsonschema>=3.0.0  # for utils
     jsonpatch>=1.15

--- a/tests/constraints.txt
+++ b/tests/constraints.txt
@@ -1,6 +1,6 @@
 # core
 scipy==1.1.0
-click==8.0.0
+click==8.0.0  # c.f. PR #1957, #1909
 tqdm==4.56.0
 jsonschema==3.0.0
 jsonpatch==1.15

--- a/tests/constraints.txt
+++ b/tests/constraints.txt
@@ -1,6 +1,6 @@
 # core
 scipy==1.1.0
-click==7.0
+click==8.0.0
 tqdm==4.56.0
 jsonschema==3.0.0
 jsonpatch==1.15

--- a/tests/constraints.txt
+++ b/tests/constraints.txt
@@ -1,6 +1,6 @@
 # core
 scipy==1.1.0
-click==8.0.0  # c.f. PR #1957, #1909
+click==8.0.0  # c.f. PR #1958, #1909
 tqdm==4.56.0
 jsonschema==3.0.0
 jsonpatch==1.15

